terraform/provider/resource_teleport_trusted_cluster.go

```diff
@@ -0,0 +1,178 @@
+// /*
+// Copyright 2015-2021 Gravitational, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/teleport-plugins/terraform/tfschema"
+	"github.com/gravitational/trace"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// resourceTeleportTrustedCluster returns Teleport github_connector resource definition
+func resourceTeleportTrustedCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTrustedClusterCreate,
+		ReadContext:   resourceTrustedClusterRead,
+		UpdateContext: resourceTrustedClusterUpdate,
+		DeleteContext: resourceTrustedClusterDelete,
+
+		Schema:        tfschema.SchemaTrustedClusterV2,
+		SchemaVersion: 2,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// resourceTrustedClusterCreate creates Teleport GithubConnector from resource definition
+func resourceTrustedClusterCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c, err := getClient(m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	n, err := getResourceName(d, "trusted_cluster")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Check if role already exists
+	_, err = c.GetRole(ctx, n)
```

Comment:
Why role? Shouldn't this check trusted cluster?

Reply:
Ouch, thanks.